MSAL/src/MSALUser.m

```diff
@@ -55,12 +74,31 @@ - (id)initWithDisplayableId:(NSString *)displayableId
         return nil;
     }
     
-    _displayableId = displayableId;
-    _name = name;
-    _identityProvider = identityProvider;
-    _uid = uid;
-    _utid = utid;
-    _environment = environment;
+    _displayableId = [displayableId copy];
+    _name = [name copy];
+    _identityProvider = [identityProvider copy];
+    _uid = [uid copy];
+    _utid = [utid copy];
+    _environment = [environment copy];
+    _account = [[MSIDAccount alloc] initWithLegacyUserId:_displayableId uniqueUserId:self.userIdentifier];
+    _account.authority = [NSURL msidURLWithEnvironment:_environment];
+    
+    return self;
+}
+
+- (instancetype)initWithAccount:(MSIDAccount *)account
+{
+    self = [self initWithDisplayableId:account.username
```

Comment:
seems a bit weird, we're using initWithAccount that calls initWithDisplayableId that creates a new account?

Reply:
yeah, the problem is that in initWithDisplayableId we create account with ids only, when initWithAccount saves account with all properties. Let me see if we can get rid of this logic.

Reply:
removed initWithAccount method.